packages/components/data-table/src/cell.js

```diff
@@ -97,6 +114,15 @@ DataCell.propTypes = {
   isCondensed: PropTypes.bool,
   isTruncated: PropTypes.bool,
   shouldIgnoreRowClick: PropTypes.bool,
+  handleRowCollapseClick: requiredIf(
+    PropTypes.func,
+    (props) => props.isTruncated
```

Comment:
Shouldn't these be _requiredIf_ `shouldRenderCollapseButton` is true, and not the `isTruncated` prop?